### PR TITLE
feat(register): added password confirmation field

### LINKED ISSUE #85: Feat: Add confirrm password field and validation to Register form
**Is your feature request related to a problem? Please describe.**
At the moment there is no confirm password field on the Register form.

**Describe the solution you'd like**
Confirm password field and its validation (checking that the password and confirm password fields match) is useful to help user confirm the password they have given on the password field. 

**Describe alternatives you've considered**
Open to contributor's idea as long as it fits the application well.

**Additional context**
NA

### PR DESCRIPTION
**Pull request by ruppysuppy**
### Description

Added password confirmation field in Regitration form

Fixes #85 

### Type of Change:

- User Interface

### How Has This Been Tested?

UI Visually tested for the confirm password field and ran automated tests

### Checklist:

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable

### CODE REVIEW

**Comment by ruppysuppy** [comment]
The build is [failing](https://travis-ci.com/github/anitab-org/bridge-in-tech-web/builds/182573315#L247), should I update the test too @mtreacy002?

**Comment by mtreacy002** [comment]
@ruppysuppy. I didn't see you put a request to work on this issue. As per the [Contributing Guideline](https://github.com/anitab-org/bridge-in-tech-web/blob/develop/.github/CONTRIBUTING.md#general-guidelines) within AnitaB.Org Open Source, you should claim an issue first and wait till a maintainer assign you to the issue. This is to prevent developers working on the same issue. It also helps keeping things in order (in case more than one developer claiming the issue, the first one who put the request is the one who normally gets to work on it). Can you imagine if you already started working on the issue before claiming it, then before you push the code someone else claim the issue and gets assign to work on it? It'll be a waste of your time, don't you think? So, for next time, I advise you to claim the issue first before start working on it. For now, can you put a request on the issue so I can assign you?

**Comment by ruppysuppy** [comment]
I would like to work on this issue

**Comment by mtreacy002** [comment]
> I would like to work on this issue

Can you post this on the issue itself (issue #85) , @ruppysuppy ?

**Comment by ruppysuppy** [comment]
Sorry 😅

**Comment by ruppysuppy** [comment]
Could you review it now @mtreacy002?

File: src/register/Register.jsx
Comment (mtreacy002) [inline src/register/Register.jsx]: It would be best to filter out the `confirmPassword` instead of adding it to the payload then delete it after. You could just add an if condition that checks if the key is confirmPassword then simply go to the next key. You can check the similar method used in PersonalDetails.jsx. Does this make sense?

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@ruppysuppy. Great job for pushing the PR pretty quickly. Can you check my feedback below? Also, it is recommended that contributors also write the test cases to test the code they are proposing. Will you be able to do this (adding another test when confirmPassword field doesn't match the password field then an error message should be shown). It's ok if you're not feeling comfortable on this, we can open another issue for another contributor to write the test cases. Let me know what you think. 😉

**Comment by ruppysuppy** [comment]
Please open another issue for it, I'm not good at writing tests, but I sure will give it a try

**Comment by mtreacy002** [comment]
> Please open another issue for it, I'm not good at writing tests, but I sure will give it a try

No worries. Will open another issue then. You've done great job with this one. Just need to address my feedback [here](https://github.com/anitab-org/bridge-in-tech-web/pull/89#discussion_r483328294) then I'll approve. Thank you for your contribution 👍.

**Comment by ruppysuppy** [comment]
I have addressed the problem. Could you review it?

**Review by mtreacy002 — DISMISSED** [review]
Good job, @ruppysuppy. LGTM now. 👍

**Comment by mtreacy002** [comment]
@anitab-org/bridgeintech-maintainers , can someone please also review this? Thanks

**Comment by ruppysuppy** [comment]
@mtreacy002 I have prepared the tests for the confirm password, I'll open a new pull request (and issue too) for it after this one is merged as that PR (test) will be dependent on this PR

**Comment by ruppysuppy** [comment]
Btw, github's saying @anitab-org/bridgeintech-maintainers is not available

**Review by mtreacy002 — APPROVED** [review]
I have approved this PR, @anitab-org/bridgeintech-maintainers and @ruppysuppy

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _Password confirmation field_  
      _Screenshot/gif/url_:  

        test-bit-web-pr-89

      _Expected Result_:  As a user, the website should have a password confirmation field on registration page.
      _Actual Result_: Same as expected

3. OS Version: Windows 10

@ruppysuppy 🎉

**Pull request merged**
